### PR TITLE
Addressed #53692 : Don't suggest extra clone when converting cloned slice to Vec

### PR DESCRIPTION
**Pull request by PramodBisht**
@sunjay @estebank  @csmoe hopefully this answer #53692
Please let me know if you have any suggestion

### CODE REVIEW

**Review by estebank — APPROVED** [review]
R=me barring the comments

File: src/librustc_typeck/check/mod.rs [hunk 1]
```diff
@@ -4726,7 +4726,21 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                         if receiver.ends_with(&method_call) {
                             None  // do not suggest code that is already there (#53348)
                         } else {
-                            Some(format!("{}{}", receiver, method_call))
+                            // methods defined will be overwrite `.clone()`
+                            // if suggesion is present in `method_call_list` (#53692)
+                            let method_call_list = [".to_vec()", ".to_string()"];
+                            if receiver.ends_with("clone()")
```
[the remark this answers is not in the document]
Comment (PramodBisht) [reply src/librustc_typeck/check/mod.rs]: done :)

File: src/test/ui/issue-53692.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+fn main() {
+        let items = vec![1, 2, 3];
+        let ref_items: &[i32] = &items;
+        let items_clone: Vec<i32> = ref_items.clone();
+
+        let s = "hi";
+        let string: String = s.clone();
```
[the remark this answers is not in the document]
Comment (PramodBisht) [reply src/test/ui/issue-53692.rs]: done!

**Comment by PramodBisht** [comment]
@estebank all suggestion addressed!

**Pull request merged**